src/Features/Core/Portable/GenerateConstructorFromMembers/AbstractGenerateConstructorFromMembersCodeRefactoringProvider.cs

```diff
@@ -47,60 +49,117 @@ protected AbstractGenerateConstructorFromMembersCodeRefactoringProvider() : this
         /// <summary>
         /// For testing purposes only.
         /// </summary>
-        protected AbstractGenerateConstructorFromMembersCodeRefactoringProvider(IPickMembersService pickMembersService_forTesting)
+        protected AbstractGenerateConstructorFromMembersCodeRefactoringProvider(IPickMembersService? pickMembersService_forTesting)
             => _pickMembersService_forTesting = pickMembersService_forTesting;
 
         protected abstract bool ContainingTypesOrSelfHasUnsafeKeyword(INamedTypeSymbol containingType);
         protected abstract string ToDisplayString(IParameterSymbol parameter, SymbolDisplayFormat format);
         protected abstract bool PrefersThrowExpression(DocumentOptionSet options);
 
-        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        public override Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            return ComputeRefactoringsAsync(context.Document, context.Span,
+                (action, applicableToSpan) => context.RegisterRefactoring(action, applicableToSpan),
+                (actions) => context.RegisterRefactorings(actions), context.CancellationToken);
+        }
+
+        public async Task<Solution?> ComputeActionForIntentAsync(Document documentBeforeIntent, TextSpan selection, Document currentDocument, string? serializedIntentData, CancellationToken cancellationToken)
+        {
+            using var builder = ArrayBuilder<CodeAction>.GetInstance(out var actions);
+            await ComputeRefactoringsAsync(documentBeforeIntent, selection, (singleAction, applicableToSpan) => actions.Add(singleAction), (multipleActions) => actions.AddRange(multipleActions), cancellationToken).ConfigureAwait(false);
+
+            if (actions.IsEmpty())
+            {
+                return null;
+            }
+
+            // The refactorings returned will be in the following order (if available)
+            // FieldDelegatingCodeAction, ConstructorDelegatingCodeAction, GenerateConstructorWithDialogCodeAction
+            // For now since we don't have an idea of which one to pick, we'll take the action that appears first in the group.
+            var codeAction = actions.First();
+            var operations = await GetCodeActionOperationsAsync(codeAction, cancellationToken).ConfigureAwait(false);
```

Comment:
This is a little fragile in that nothing is preventing this provider from changing the actions.  One idea I had was to create something like a SolutionChangesCodeAction that enforces only ApplyChangesOperations can be returned.  We'd have to do something similar for code actions with options (e.g. once given options, it only returns applychangesoperations).   But I'm open to better ideas